packages/dd-trace/src/appsec/api_security_sampler.js

```diff
@@ -1,61 +1,84 @@
 'use strict'
 
+const TTLCache = require('@isaacs/ttlcache')
+const web = require('../plugins/util/web')
 const log = require('../log')
+const { AUTO_REJECT, USER_REJECT } = require('../../../../ext/priority')
+
+const MAX_SIZE = 4096
 
 let enabled
-let requestSampling
+let sampledRequests
 
-const sampledRequests = new WeakSet()
+class NoopTTLCache {
+  clear () { }
+  set (key) { return undefined }
+  has (key) { return false }
+}
 
 function configure ({ apiSecurity }) {
   enabled = apiSecurity.enabled
-  setRequestSampling(apiSecurity.requestSampling)
+  sampledRequests = apiSecurity.sampleDelay === 0
+    ? new NoopTTLCache()
+    : new TTLCache({ max: MAX_SIZE, ttl: apiSecurity.sampleDelay * 1000 })
```

Comment:
since in config you do `maybeFloat()` to parse this field, if it's not a number, it will return undefined. What will this code do if sampleDelay is undefined ?

Reply:
I think it will be NaN because undefined * 1000 is NaN

Reply:
from TTLCache code
```javascript
    if (ttl !== undefined && !isPosIntOrInf(ttl)) {
      throw new TypeError(
        'ttl must be positive integer or Infinity if set'
      )
    }
```

Reply:
Should we let the startup throw ? or should we ignore with a default value ?

Reply:
ok apparently it takes the default value from config.js